WalletWasabi/Blockchain/Keys/KeyState.cs

```diff
@@ -4,5 +4,6 @@ public enum KeyState
 {
 	Clean,
 	Locked,
-	Used
+	Used,
+	Obsolete
```

Comment:
Lifecycle of a Key:

1. Clean
2. Locked
3. Used <= script appeared on the output side of a transaction
4. Spent <= script appeared on the input side of the transaction

I suggest to rename Obsolete to Spent.